src/dll/edbgEcmdDll.C

```diff
@@ -1696,11 +1702,269 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
 #endif // ECMD_REMOVE_MEMORY_FUNCTIONS
 
 #ifndef ECMD_REMOVE_INIT_FUNCTIONS
+
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
+
+static edbgIPLTable g_edbgIPLTable;
+
+/**
+  * @brief Set the IPL mode
+  *
+  * @param enum ipl_mode i_mode - IPL mode (Interactive or Continous IPL mode)
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t setIplMode(enum IplMode_t i_mode)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    
+    /* If IPL mode is set to interactive then, SBE will be set in ISTEP mode
+     * and execute all SBE isteps in chip op mode */
+    if (i_mode == IPL_MODE_ISTEP)
+    {
+        ecmdChipTarget target;
+        constexpr uint32_t MBOX_SCRATCH_REG3 = 0x283a;
+        constexpr uint32_t MBOX_SCRATCH_REG8 = 0x283f;
+        ecmdDataBuffer dataBuffer;
+
+        target.chipTypeState = ECMD_TARGET_FIELD_VALID;
+        target.cageState = target.nodeState = target.slotState = target.posState = ECMD_TARGET_FIELD_VALID;
+        target.cage = target.node = target.slot = 0;
+
+        //This will be actually set by sbe_config_update istep procedure.
+        //For testing purpose we are setting it up.
+        dataBuffer.setWordLength(1);
+        dataBuffer.setWord(0, 0x86000000);
```

Comment:
0x80000000 is good enough to set in istep mode.

Reply:
Ok. well 0x86000000 is still fine. I just did OR with default value currently, Witherspoon has. currently, edbg has putcfam under mask is not implemented. will need to use for future enhancement so value is OR of existing settings